### ISSUE TITLE
failed to allocate tensors on mobile device

### ISSUE THREAD

**Issue by guy6656** [tensorflow/tensorflow#18890]
### System information
I minimize the model to very simple model with one input, convolution layer and pool is the output.
I work on Ubuntu 16.04
I compile TensorFlow Lite from source with Bazel
TensorFlow version is 1.8
Python 3.6
Bazel version 0.10.0
GCC/Compiler version - gcc (Ubuntu 5.4.0-6ubuntu1~16.04.6) 5.4.0 20160609
Working on CPU

### Describe the problem
I am trying to run my Tensorflow model on samsung galaxy s9, 
I trained model with TensorFlow freeze it and convert it to .tflite model file with toco.
I am running CPP program on the phone according to documentation and load the .tflite model file:

    std::unique_ptr<tflite::FlatBufferModel> model(tflite::FlatBufferModel::BuildFromFile(modelFile));
    if (!model) {
        printf("Failed to mmap model %s\n", g_modelFile);
    }
    model->error_reporter();
    tflite::ops::builtin::BuiltinOpResolver resolver;
    std::unique_ptr<tflite::Interpreter> interpreter;
    tflite::InterpreterBuilder(*model, resolver)(&interpreter);
    if (!interpreter) {
        printf("Failed to construct interpreter\n");
    }
    interpreter->UseNNAPI(false);
    if (interpreter->AllocateTensors() != kTfLiteOk) {
        printf("Failed to allocate tensors!\n");
    }

I succeed to initialize the model but failed to run **interpreter->AllocateTensors()**
There is no print of why it failed to allocate the tensors, just my print that the function failed

    if (interpreter->AllocateTensors() != kTfLiteOk) {
        printf("Failed to allocate tensors!\n");
    }

I minimized my model to include only reshape, convolution layer and pool and still failed to run AllocateTensors

Output with debug information:

resolved reporter
tensors size: 8
nodes size: 3
inputs: 1
input(0) name: patches
input(1) name: (null)
Reshape, 237552, 1, 0.000000, 0
Reshape/shape, 16, 2, 0.000000, 0
conv1/Relu, 1861632, 1, 0.000000, 0
conv1/convolution_bias, 128, 1, 0.000000, 0
conv1/kernel, 3456, 1, 0.000000, 0
input/patches-input, 237552, 1, 0.000000, 0
output, 465408, 1, 0.000000, 0
patches, 4, 1, 0.000000, 0

**Failed to allocate tensors!**

**Comment by guy6656** [comment]
Hi,

After some debugging I found the error:
tensorflow/contrib/lite/simple_memory_arena.cc:82 it->size != alloc.size (4 != 0)

It happens on the first allocation,

The initialization succeeded but it fails with this allocation, can you please give me some directions where to look?

My model is very basic and consists weights and biases:

# Create model
def my_model(patches, hidden_init):

    patches = tf.reshape(patches, (101, 14, 14, 3))
    hidden_init = tf.reshape(hidden_init, shape=[1, 512])

    conv1 = tf.layers.conv2d(patches, 32, [3, 3], activation=tf.nn.relu, name='conv1')
    pool1 = tf.layers.max_pooling2d(conv1, 2, 2)

    output = tf.identity(pool1, 'output')

Thanks

**Comment by guy6656** [comment]
Hi,

Another thing I tried is to run Mobilenet model for tflite on the phone and it worked and succeeded to allocate the tensors.

Maybe there is a problem on the way I have converted the files with Toco?
(This is the command I ran, there was no errors when I used the following)
bazel-bin/tensorflow/contrib/lite/toco/toco 
--input_format=TENSORFLOW_GRAPHDEF 
--input_file=/home/model_for_freezing/output_graph.pb 
--output_format=TFLITE 
--output_file=/tmp/my_model.tflite 
--inference_type=FLOAT 
--inference_input_type=FLOAT 
--input_arrays=patches
--output_arrays=output                   

Thanks

**Comment by cogmeta** [comment]
I have similar problem. 

Loaded model foo.tflite
resolved reporter
construct interpreter
tensors size: 15
nodes size: 5
inputs: 1
input(0) name: input
0: MatMul_bias, 48, 1, 0, 0
1: Reshape, 15680, 1, 0, 0
2: add, 48, 1, 0, 0
3: fc1/MatMul_bias, 512, 1, 0, 0
4: fc1/Relu, 512, 1, 0, 0
5: fc1/W/transpose, 2007040, 1, 0, 0
6: fc2/MatMul_bias, 512, 1, 0, 0
7: fc2/Relu, 512, 1, 0, 0
8: fc2/W/transpose, 65536, 1, 0, 0
9: fc3/MatMul_bias, 512, 1, 0, 0
10: fc3/Relu, 512, 1, 0, 0
11: fc3/W/transpose, 65536, 1, 0, 0
12: final_fc/transpose, 6144, 1, 0, 0
13: input, 15840, 1, 0, 0
14: labels_softmax, 48, 1, 0, 0
input: 13
number of inputs: 1
number of outputs: 1
tensorflow/contrib/lite/simple_memory_arena.cc:82 it->size != alloc.size (15680 != 0)
Failed to allocate tensors!Segmentation fault

**Comment by cogmeta** [comment]
though, i am trying on raspberry pi.

**Comment by cogmeta** [comment]
Linking to https://github.com/tensorflow/tensorflow/issues/19026. Both issues probably have same cause.

**Comment by cogmeta** [comment]
My guess is The error is happening on first allocation. It is trying to deallocate a tensor which has not been allocated yet.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 30 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 45 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 60 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by aselle** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
